### PR TITLE
converting shell file to python with latest modification

### PR DESCRIPTION
**Pull request by Naresh-ibm**
re-writting the script from shell to python. handled the
dependency module unload without giving them in config file.
for FC adapter multipath dependency is handled properly.
code can be run for a single module or for all the driver
modules available in the OS

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[module_load_unload_v2_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3622464/module_load_unload_v2_job_log.txt)

File: io/driver/module_unload_load.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    def setUp(self):
+        """
+        get parameters.
+        """
+        self.module = self.params.get('module', default=None)
+        self.iteration = self.params.get('iteration', default=1)
+        self.only_io = self.params.get('only_io', default=None)
+        # self.fc = self.params.get('fc', default=None)
+        self.load_unload_sleep_time = 30
+        self.multipath_sleep_time = 5
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: @Naresh-ibm recently, I had faced similar issue of removing multipath before reloading module, and handled it via:
```
# cat module_reload.sh 
#!/usr/bin/bash

for i in {1..20}
do
    while [[ `lsmod | grep ^mpt3sas | awk '{print $NF}'` != '0' ]]
    do
        multipath -F
        sleep 5
    done
        lsmod | grep ^mpt3sas
    echo "Iteration $i"
    echo "------------"
    rmmod mpt3sas && sleep 60
    modprobe mpt3sas && sleep 300
    dmesg -c
    echo "============================================"
done
```
Please see if this approach is helpful here

Comment (narasimhan-v) [reply io/driver/module_unload_load.py]: or use `wait_for()`

File: io/driver/module_unload_load.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join("module_unload_load.py.data/config")
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: please check this once

File: io/driver/module_unload_load.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        self.load_unload_sleep_time = 30
+        self.multipath_sleep_time = 5
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        uname = linux_modules.platform.uname()[2]
+        # cmd = "cat /lib/modules/%s/modules.builtin" % uname
+        # for each in str(process.getoutput(cmd)).split('\n'):
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: comments

File: io/driver/module_unload_load.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    :only_io True for single provide module and False for All pci modules
+    """
+    def setUp(self):
+        """
+        get parameters.
+        """
+        self.module = self.params.get('module', default=None)
+        self.iteration = self.params.get('iteration', default=1)
+        self.only_io = self.params.get('only_io', default=None)
+        # self.fc = self.params.get('fc', default=None)
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: comments

File: io/driver/module_unload_load.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join("module_unload_load.py.data/config")
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                module1 = line.split('=')[-1]
+                break
+        return module1
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: please check the scope

File: io/driver/module_unload_load.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    sub_mod = sub_mod.split(' ')
+                    for mod in sub_mod:
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: `for mod in sub_mod.split()`

**Comment by Naresh-ibm** [comment]
created new PR with suggested changes.
hence closing in favour of #1296

**Pull request closed without merging**